tests/suites/test_suite_gcm.function

```diff
@@ -431,6 +431,30 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void gcm_update_output_buffer_too_small( int cipher_id, int mode,
+                                         data_t * key_str, const data_t *input,
+                                         const data_t *iv )
+{
+    mbedtls_gcm_context ctx;
+    uint8_t *output = NULL;
+    size_t olen;
+    size_t output_len = input->len - 1;
+
+    mbedtls_gcm_init( &ctx );
+    TEST_EQUAL( mbedtls_gcm_setkey( &ctx, cipher_id, key_str->x, key_str->len * 8 ), 0 );
+    TEST_EQUAL( 0, mbedtls_gcm_starts( &ctx, mode, iv->x, iv->len ) );
+
+    ASSERT_ALLOC( output, output_len );
+    olen = 0xdeadbeef;
```

Comment:
Is it necessary to set it?

Reply:
Setting `olen` to a “nonsense” is useful if we test it after the library function returns. It allows the test to validate that the library set it to the desired value.

In this case, the test expects the library function to fail, and the documentation does not guarantee that `olen` is set if `mbedtls_gcm_update` fails. In the low-level crypto API, we don't particularly design functions to be robust to misuse, and there are no documented expectations on alternative implementations, so there's nothing the test can check about `olen` on failure. There are places in the PSA crypto tests where we do test `olen`-like output parameters on failure because this is a higher-level API that aims to be robust to misuse.

Reply:
The reason for my question was to determine the purpose of this line. It's irrelevant here, we're not checking it, so I would suggest simply dropping it so that the test case is minimal.